Utilities/AzureOnboardingCLI/azure-automatic-role-assignment.py

```diff
@@ -71,20 +82,34 @@ def get_or_create_spot_account(subscription_id, name, token):
     Returns:
         str: The ID of the created account.
     """
-    session = SpotinstSession(auth_token=token, base_url=SPOT_API_BASE_URL)
-    client = session.client("admin")
+    try:
+        session = SpotinstSession(auth_token=token, base_url=SPOT_API_BASE_URL)
+    except SpotinstClientException as e:
+        log(f"Spotinst token is invalid.", LogLevel.ERROR)
+        raise e
+
+    client = session.client("admin", timeout=15)
+
+    try:
+        existing_accounts = [account["account_id"] for account in client.get_accounts() if
+                             account["provider_external_id"] == subscription_id and account["cloud_provider"] == 'AZURE']
+    except ConnectTimeout as e:
+        log(f"Could not reach Spot API. Please try again later.", LogLevel.ERROR)
```

Comment:
Yes and no.  I noticed it when I had a bad local configuration.  But the experience is bad enough that it will sit and wait for minutes.  So I figured it could happen, and if it did it's easy enough to account for it.